### PR TITLE
refactor and add k8s support

### PR DESCRIPTION
**Pull request by aisenbergmor**

### CODE REVIEW

File: rbz_api_tester/Executers/ApiExecuter.py [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+                final_api_url, headers=headers, params=params, verify=False
+            )
+        elif method == "PATCH":
+            response = requests.patch(
+                final_api_url, headers=headers, data=params, files=files, verify=False
+            )
+        elif method == "OPTIONS":
+            response = requests.options(final_api_url, headers=headers, verify=False)
+        elif method == "HEAD":
+            response = requests.head(final_api_url, headers=headers, verify=False)
```
[the remark this answers is not in the document]
Comment (aisenbergmor) [reply rbz_api_tester/Executers/ApiExecuter.py:196]: #jit_ignore_accept
send requests to test environment to simulate traffic

File: rbz_api_tester/Executers/ApiExecuter.py [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+        self._add_files(files)
+        final_api_url = f"https://{api}"
+        CommonParameters.logger.debug(f"\t\tCalling: {final_api_url}")
+        if method == "GET":
+            response = requests.get(
+                final_api_url, headers=headers, params=params, verify=False
+            )
+        elif method == "POST":
+            response = requests.post(
+                final_api_url, headers=headers, data=params, files=files, verify=False
```
[the remark this answers is not in the document]
Comment (aisenbergmor) [reply rbz_api_tester/Executers/ApiExecuter.py:179]: #jit_ignore_accept
send requests to test environment to simulate traffic

File: rbz_api_tester/Executers/ApiExecuter.py [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+            )
+        elif method == "PUT":
+            response = requests.put(
+                final_api_url, headers=headers, data=params, files=files, verify=False
```
[the remark this answers is not in the document]
Comment (aisenbergmor) [reply rbz_api_tester/Executers/ApiExecuter.py:183]: #jit_ignore_accept
send requests to test environment to simulate traffic

File: rbz_api_tester/Executers/ApiExecuter.py [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+        elif method == "DELETE":
+            response = requests.delete(
+                final_api_url, headers=headers, params=params, verify=False
+            )
+        elif method == "PATCH":
+            response = requests.patch(
+                final_api_url, headers=headers, data=params, files=files, verify=False
+            )
+        elif method == "OPTIONS":
+            response = requests.options(final_api_url, headers=headers, verify=False)
```
[the remark this answers is not in the document]
Comment (aisenbergmor) [reply rbz_api_tester/Executers/ApiExecuter.py:194]: #jit_ignore_accept
send requests to test environment to simulate traffic

File: rbz_api_tester/Executers/ApiExecuter.py [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+        files = {}
+        CommonParameters.logger.debug(f"\t\tMethod is {method}")
+        self._add_custom_headers(headers)
+        self._add_arguments(params)
+        self._add_files(files)
+        final_api_url = f"https://{api}"
+        CommonParameters.logger.debug(f"\t\tCalling: {final_api_url}")
+        if method == "GET":
+            response = requests.get(
+                final_api_url, headers=headers, params=params, verify=False
```
[the remark this answers is not in the document]
Comment (aisenbergmor) [reply rbz_api_tester/Executers/ApiExecuter.py:175]: #jit_ignore_accept
send requests to test environment to simulate traffic

File: rbz_api_tester/Executers/PythonExecuter.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        )
+        CommonParameters.logger.debug(f"\t\tPython Script:\n{code}")
+
+        process = subprocess.Popen(
+            ["python"],
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+            shell=True,
```
[the remark this answers is not in the document]
Comment (aisenbergmor) [reply rbz_api_tester/Executers/PythonExecuter.py:27]: #jit_ignore_accept
execution of a python script for test purposes during automation run

File: rbz_api_tester/Executers/ApiExecuter.py [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+            response = requests.post(
+                final_api_url, headers=headers, data=params, files=files, verify=False
+            )
+        elif method == "PUT":
+            response = requests.put(
+                final_api_url, headers=headers, data=params, files=files, verify=False
+            )
+        elif method == "DELETE":
+            response = requests.delete(
+                final_api_url, headers=headers, params=params, verify=False
```
[the remark this answers is not in the document]
Comment (aisenbergmor) [reply rbz_api_tester/Executers/ApiExecuter.py:187]: #jit_ignore_accept
send requests to test environment to simulate traffic

File: rbz_api_tester/Executers/ApiExecuter.py [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+            )
+        elif method == "PATCH":
+            response = requests.patch(
+                final_api_url, headers=headers, data=params, files=files, verify=False
```
[the remark this answers is not in the document]
Comment (aisenbergmor) [reply rbz_api_tester/Executers/ApiExecuter.py:191]: #jit_ignore_accept
send requests to test environment to simulate traffic

**Review by olgasobolevsky — APPROVED** [review]

**Pull request merged**